### PR TITLE
mas 1.4.3

### PR DESCRIPTION
**Pull request by phatblat**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This is a complete rework of the formula after #31058 was unsuccessful. I'm the maintainer for `mas` and will build custom bottles for older macOS versions once this PR is accepted.

### CODE REVIEW

File: Formula/mas.rb [hunk 1]
```diff
@@ -1,25 +1,40 @@
...
+    # :revision => "966872b32820c014a9004691f5da47f170702236",
+    :shallow => true
+  head "https://github.com/mas-cli/mas.git",
+    :shallow => true
 
   bottle do
     cellar :any_skip_relocation
     sha256 "fb05286adfb9ab0a63a8027391647fb874fa810c07d412505f5bc7911139a3f2" => :mojave
     sha256 "b7585ced3a93d60e95357e93d729913a6f628fda82359e77c6553c2e802c50dc" => :high_sierra
-    sha256 "af5be6aa9902d9cfc2aa69dbf313441a7c201463d516face721f900ceae9556b" => :sierra
```
Comment (commitay) [inline Formula/mas.rb]: The bottle block shouldn't be edited manually.

File: Formula/mas.rb [hunk 2]
```diff
@@ -1,25 +1,40 @@
...
   desc "Mac App Store command-line interface"
   homepage "https://github.com/mas-cli/mas"
-  url "https://github.com/mas-cli/mas/archive/v1.4.1.tar.gz"
-  sha256 "4fd91c13b46d403b52dbee3891adb3cd6571e07ad20cf58de0100c9f695e6c24"
-  head "https://github.com/mas-cli/mas.git"
+  url "https://github.com/mas-cli/mas.git",
+    #:branch => "release-1.4.3",
+    :tag => "v1.4.3",
+    # :revision => "966872b32820c014a9004691f5da47f170702236",
+    :shallow => true
```
Comment (commitay) [inline Formula/mas.rb]: `shallow => true` is redundant.

File: Formula/mas.rb [hunk 3]
```diff
@@ -1,25 +1,40 @@
...
   end
 
-  depends_on :xcode => ["9.0", :build]
+  depends_on "carthage" => :build
+  depends_on :xcode => ["10.0", :build]
 
   def install
+    # Prevent build failures from warnings
+    xcconfig = buildpath/"Overrides.xcconfig"
+    File.open(xcconfig, "w") { |file| file.write("GCC_TREAT_WARNINGS_AS_ERRORS = NO") }
```
Comment (commitay) [inline Formula/mas.rb]:
```suggestion
    xcconfig.write("GCC_TREAT_WARNINGS_AS_ERRORS = NO")
```

File: Formula/mas.rb [hunk 4]
```diff
@@ -1,8 +1,8 @@
 class Mas < Formula
   desc "Mac App Store command-line interface"
   homepage "https://github.com/mas-cli/mas"
-  url "https://github.com/mas-cli/mas/archive/v1.4.1.tar.gz"
-  sha256 "4fd91c13b46d403b52dbee3891adb3cd6571e07ad20cf58de0100c9f695e6c24"
+  url "https://github.com/mas-cli/mas.git",
+    :tag => "v1.4.3",
```
Comment (fxcoudert) [inline Formula/mas.rb]: Extra comma

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/mas.rb (hunk 4 above)
Comment (phatblat) [reply Formula/mas.rb]: Ack! Thx
```suggestion
    :tag => "v1.4.3"
```

File: Formula/mas.rb [hunk 5]
```diff
@@ -1,8 +1,8 @@
 class Mas < Formula
   desc "Mac App Store command-line interface"
   homepage "https://github.com/mas-cli/mas"
-  url "https://github.com/mas-cli/mas/archive/v1.4.1.tar.gz"
-  sha256 "4fd91c13b46d403b52dbee3891adb3cd6571e07ad20cf58de0100c9f695e6c24"
+  url "https://github.com/mas-cli/mas.git",
+    :tag => "v1.4.3"
```
Comment (commitay) [inline Formula/mas.rb]: `revision` is required for stable.

```diff
diff --git a/Formula/mas.rb b/Formula/mas.rb
index ca67eaf853..b0b6159f7b 100644
--- a/Formula/mas.rb
+++ b/Formula/mas.rb
@@ -2,7 +2,8 @@ class Mas < Formula
   desc "Mac App Store command-line interface"
   homepage "https://github.com/mas-cli/mas"
   url "https://github.com/mas-cli/mas.git",
-    :tag => "v1.4.3"
+      :tag => "v1.4.3",
+      :revision => "11a0e3e14e5a83aaaba193dfb6d18aa49a82b881"
   head "https://github.com/mas-cli/mas.git"
 
   bottle do
```

File: Formula/mas.rb [hunk 6]
```diff
@@ -12,14 +13,26 @@ class Mas < Formula
...
+    ENV["XCODE_XCCONFIG_FILE"] = xcconfig
+
+    system "carthage", "bootstrap", "--platform", "macOS"
+
     xcodebuild "-project", "mas-cli.xcodeproj",
                "-scheme", "mas-cli Release",
                "-configuration", "Release",
-               "SYMROOT=build"
-    bin.install "build/mas"
+               "OBJROOT=#{buildpath.realpath}",
```
Comment (commitay) [inline Formula/mas.rb]: Why is `buildpath.realpath` needed?

**Comment by commitay** [comment]
@BrewTestBot test this please

**Comment by phatblat** [comment]
@fxcoudert This should be GTG now. I had to work around some linker errors with `-headerpad_max_install_names`

File: Formula/mas.rb [hunk 7]
```diff
@@ -12,14 +13,28 @@ class Mas < Formula
...
-  depends_on :xcode => ["9.0", :build]
+  depends_on "carthage" => :build
+  depends_on :xcode => ["10.0", :build]
 
   def install
-    xcodebuild "-project", "mas-cli.xcodeproj",
+    # Prevent warnings from causing build failures
+    # Prevent linker errors by telling all lib builds to use max size install names
+    xcconfig = buildpath/"Overrides.xcconfig"
+    xcconfig.write("GCC_TREAT_WARNINGS_AS_ERRORS = NO\nOTHER_LDFLAGS = -headerpad_max_install_names")
```
Comment (commitay) [inline Formula/mas.rb]:
```
xcconfig.write <<~EOS
  GCC_TREAT_WARNINGS_AS_ERRORS = NO
  OTHER_LDFLAGS = -headerpad_max_install_names
EOS
```

Comment (phatblat) [reply Formula/mas.rb]: 👍

**Review by commitay — APPROVED** [review]

**Comment by phatblat** [comment]
Thanks for your feedback @commitay and @fxcoudert ! 🙇

**Comment by phatblat** [comment]
Thanks for your feedback @commitay and @fxcoudert! 🙇

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thanks @phatblat for the pull request

**Pull request closed without merging**